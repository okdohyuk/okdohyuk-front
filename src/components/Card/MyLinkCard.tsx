import React from 'react';
import Link from '@components/Link';

type LinkCard = {
  title: string;
  explanation: string;
  link: string;
};

function MyLinkCard({ link, explanation, title }: LinkCard) {
  return (
    <Link href={link} hasTargetBlank>
      <article className="rounded-lg p-4 md:p-8 transition-all bg-zinc-300 dark:bg-zinc-800 shadow-xl dark:shadow-lg text-cyan-500 border-2 border-transparent hover:border-current">
        <div className="flex gap-2 justify-between items-center">
          <div className="text-2xl font-extrabold ">{title}</div>
        </div>
        <div className="text-sm mt-2 text-black dark:text-white">{explanation}</div>
      </article>
    </Link>
  );
}

export default MyLinkCard;
