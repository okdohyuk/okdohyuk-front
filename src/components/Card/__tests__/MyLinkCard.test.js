import React from 'react';
import { render } from '@testing-library/react';
import MyLinkCard from '@components/Card/MyLinkCard';

describe('<MyLinkCard />', () => {
  it('matches snapshot', () => {
    const utils = render(
      <MyLinkCard title={'제목'} explanation={'테스트 설명입니다.'} link={'/'} />,
    );
    expect(utils.container).toMatchSnapshot();
  });
});
