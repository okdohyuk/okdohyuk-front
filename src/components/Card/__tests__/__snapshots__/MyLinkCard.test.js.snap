// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyLinkCard /> matches snapshot 1`] = `
<div>
  <a
    href="/"
    target="_blank"
  >
    <article
      class="rounded-lg p-4 md:p-8 transition-all bg-zinc-300 dark:bg-zinc-800 shadow-xl dark:shadow-lg text-cyan-500 border-2 border-transparent hover:border-current"
    >
      <div
        class="flex gap-2 justify-between items-center"
      >
        <div
          class="text-2xl font-extrabold "
        >
          제목
        </div>
      </div>
      <div
        class="text-sm mt-2 text-black dark:text-white"
      >
        테스트 설명입니다.
      </div>
    </article>
  </a>
</div>
`;
